src/components/starlight/MarkdownContent.astro

```diff
@@ -0,0 +1,20 @@
+---
+import type { Props } from '@astrojs/starlight/props';
+import '@astrojs/starlight/style/markdown.css';
+
+import { Aside } from '@astrojs/starlight/components';
+import { coerce, satisfies } from 'semver'
+
+const { astroRange } = Astro.props.entry.data;
+---
+
+{
+  // only show if the range is specified, and is not satisfied by the latest version
+  astroRange && !satisfies(Astro.locals.astroLatest, astroRange) && (
+    <Aside type="caution">
+      This guide was written for Astro {coerce(astroRange)} and may not work with more recent versions of Astro.
+    </Aside>
+  )
+}
```

Comment:
nit: I like a more _"classical"_ style using `if () {}` in this case. It makes it more understandable.

Reply:
I don't think you _can_ add an if block in an Astro template

Reply:
True.. I missed that. You can add them in a `.map(() => {})`, but you can't directly in Astro's "jsx"..